django/db/backends/mysql/validation.py

```diff
@@ -37,8 +37,8 @@ def check_field_type(self, field, field_type):
         if (field_type.startswith('varchar') and field.unique and
                 (field.max_length is None or int(field.max_length) > 255)):
             errors.append(
-                checks.Error(
-                    'MySQL does not allow unique CharFields to have a max_length > 255.',
+                checks.Warning(
+                    'CharField  have length > 255.',
```

Comment:
We can use `display_name` instead of `MySQL`. I would keep the previous wording, e.g.:
```
%s may not allow unique CharFields to have a max_length > 255.
```